src/context.rs

```diff
@@ -653,6 +653,77 @@ impl Context {
         }
     }
 
+    pub fn activate_credential(
+        &mut self,
+        activate_handle: KeyHandle,
+        key_handle: KeyHandle,
+        credential_blob: IDObject,
+        secret: EncryptedSecret,
+    ) -> Result<Digest> {
+        let mut out_cert_info = null_mut();
+        let ret = unsafe {
+            Esys_ActivateCredential(
+                self.mut_context(),
+                activate_handle.into(),
+                key_handle.into(),
+                self.required_session_1()?,
+                self.required_session_2()?,
+                self.optional_session_3(),
+                &credential_blob.try_into()?,
+                &secret.try_into()?,
+                &mut out_cert_info,
+            )
+        };
+
+        let ret = Error::from_tss_rc(ret);
+
+        if ret.is_success() {
+            let out_cert_info = unsafe { MBox::<TPM2B_DIGEST>::from_raw(out_cert_info) };
+
+            Ok(Digest::try_from(*out_cert_info)?)
+        } else {
+            Err(ret)
```

Comment:
Could you add an `error` message here? 😇

Reply:
Hah, will do.